### ISSUE TITLE
Suggestion: A litter replaces existing one on same square.

### ISSUE THREAD

**Issue by Glimmer-Man** [CorsixTH/CorsixTH#581]
It would be more logic, that litter of same type will replace another one on same square. Vomit over vomit makes one bigger vomit, but not two layers of vomits.

It's weird that handyman cleans one place again and again.

And I think it will solve #388.

**Comment by lewri** [comment]
On checking the original, only 1 litter should be on a tile.

Vomit trumps regular litter, so if someone vomits on e.g. a banana the banana gets removed.
If someone vomits on a vomit it consists as just one vomit.

**Comment by ARGAMX** [comment]
Related to #2759

**Comment by lewri** [comment]
A hierarchy of litter types needs to be confirmed in the original before implementing this.

Currently we have the following relevant litter types
```lua
litter_types["pee"] = 2248
litter_types["dead_rat"] = 2242
litter_types["puke"] = 2060
litter_types["soda_can"] = 1894
litter_types["banana"] = 1896
litter_types["paper"] = 1898
litter_types["bottle"] = 1900
```

**Comment by ARGAMX** [comment]
<img width="912" alt="Image" src="https://github.com/user-attachments/assets/2e9b5224-8e28-408e-95bb-8cc1abbc18ff" />

[Litter placement research]

So, I did some research on litter and to summarize:

1. There can only be 1 litter per square. No any layers per title. (Tested)
2. Killing a mouse always replaces any other litter of all types in that square with blood. (Tested)
3. Vomit always replaces any other litter in that square, including blood. (Tested)

So blood and vomit always replace each other depending on what happened later. (Tested)

4. General litter (can, banana, trash) never replaces any other type of litter. Not even another General litter. (Tested)
5. General litter can be generated only on a clean title. (Tested)

I was unable to experiment with pee. 
But I think it can be assumed that pee has the same rules as blood and vomit - it replaces blood and vomit if it happen later. As TheCycoONE wisely noticed, we can stick with this rule until someone prove otherwise.

**Comment by ARGAMX** [comment]
Also, due to absence of stacking litter in same title in vanilla TH handyman fully clears title in 1 attempt. 

In CorsixTH currently if there several layers of litter he will clean only ONE. 
Next, he leaving this not completely cleaned title if there any other job for him!

https://github.com/user-attachments/assets/649df243-e3dc-45f1-bcff-424e12f83127

**Comment by Flavsditz** [comment]
I think I have an idea on how to tackle this one but wanted to make sure the preffered solution is the non-stacking of litter. 

I ask because the other solution could be to simply clean all the litter in the tile using `World:removeAllLitter()` instead of `remove()` which as far as I can see is currently used. 
Using the solution mentioned however would not solve the problem of a possible stack overflow issue per #2759, so I would tend to avoid the stacking

**Comment by ARGAMX** [comment]
> I think I have an idea on how to tackle this one but wanted to make sure the preffered solution is the non-stacking of litter.
> 
> I ask because the other solution could be to simply clean all the litter in the tile using `World:removeAllLitter()` instead of `remove()` which as far as I can see is currently used. Using the solution mentioned however would not solve the problem of a possible stack overflow issue per [#2759](https://github.com/CorsixTH/CorsixTH/issues/2759), so I would tend to avoid the stacking

Yes, non-stacking of litter is acceptable solution as it follows original.

I like the both solutions non-stacking of litter and simply clean all the litter.
Although simply clean all the litter does not solve problem 2759.

I even tried `World:removeAllLitter()` solution at some point.
But when I implemented it and started testing it, I noticed that each litter layer creates its own separate task in the task manager for handymen. 

Thus, if there are 2 pieces of litter in a tile, hypothetically, 2 handymen would go to it, each to clean up their own litter.

But after `World:removeAllLitter()`call, the task of the second one handyman is removed from the task manager. Since the first handyman has already cleaned everything up. So the second one handyman was coming to that tile in vain.

So there shouldn’t just call `World:removeAllLitter()`, but rather ensure that the task manager for the handyman doesn’t create new tasks when new litter is added to the same tile. Otherwise, a new defect in logic is obtained.

So, it turned out that this solution is not as simple as it seems at first glance and requires even more effort. Also it do not solve 2759. At this stage, I switched to another task, and haven't continued work on this task.

**Comment by Flavsditz** [comment]
Welcoming feedback on the PR attempting to implement this one
